Change list handling in ProgramOptions

jmini opened the pull request.

selesse commented on jxlint/src/main/java/com/selesse/jxlint/model/ProgramOptions.java:
```diff
@@ -114,30 +140,109 @@ public void setSourceDirectory(String sourceDirectory) {
      * given a raw option string "Rule1, Rule2" (i.e. parsed from <code>--disable "Rule1, Rule2"</code>), this returns
      * a list with { "Rule1", "Rule2" }. If any of the rules passed as strings do not exist, this will function
      * will throw a {@link com.selesse.jxlint.model.rules.NonExistentLintRuleException}.
      */
```
This Javadoc needs to be updated to reflect the new signature.

jmini commented:
I have updated my branch and I have added additional javadoc.

I have no idea how I can remove or mark as DONE this "Changes requested" status in the Pull Request.
